src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -2065,45 +2065,75 @@ object FunctionRegistry {
     """, "n" -> "Number of items to take.")(
     aggregableHr(TTHr), intHr, arrayHr(TTHr))
 
-  registerLambdaAggregator("takeBy", (f: (Any) => Any, n: Int) => new TakeByAggregator[Int](f, n),
+  private val genericTakeByDocs = """
+    Returns the first ``n`` items of an aggregable in ascending order, ordered
+    by the result of ``f``. ``NA`` always appears last. If the aggregable
+    contains less than ``n`` items, then the result will contain as many
+    elements as the aggregable contains.
+
     """
-    Take the first ``n`` items of an aggregable ordered by the result of ``f``.
 
+  private val integralTakeByDocs = genericTakeByDocs ++ """
     **Examples**
 
-    Returns the 10 samples with the largest number of singletons:
+    Consider an aggregable ``gs`` containing these elements::
+
+      7, 6, 3, NA, 1, 2, NA, 4, 5, -1
```

Comment:
Yeah the `::` followed by two new lines and then indented stuff becomes a code block.